submissions/SamVal007/Friends_APP/script.js

```diff
@@ -0,0 +1,100 @@
+const CONTAINER = document.getElementById('container');
+const SIDEBAR = document.getElementById('sidebar');
+const SEARCH_FIELD = document.getElementById('searchField');
+const URL = 'https://randomuser.me/api/?results=40';
+
+let persons = [];
+let displayList = [];
+
+function getPeopleData() {
+  let response = fetch(URL)
+    .then(
+      successResponse => {
+        if (successResponse.status != 200) {
+          return null;
+        } else {
+          return successResponse;
+        }
+      }
+    )
+    .then(data => data.json())
+    .then(function (data) {
+      persons = data.results;
+      displayList = persons;
+      displayCards();
```

Comment:
Single responsibility - your functions should do only one job. As an example function in which you fetch users should only fetch them and not render, transform or process them in other ways. All these things should be done in another place, outside your function. The same applied to the sort function, which usually does all types of sorting 😉

Reply:
I would be honest, but i didnt find the way how to delete this displayCards() from this piece of code, so i just renamed function. I was trying to put this  displayCards() in document.addEventListener('DOMContentLoaded', function () , but it didnt work...

Reply:
Tip - you should use `then` in special place ;)

Reply:
Yep, this I understood, but where:)
Ok, I will try to figure out!

Reply:
"in special place" 😄

Reply:
Ok, I think I found out how to do it!
First of all - I tried to use one more **promise** inside of  `document.addEventListener('DOMContentLoaded', function (){...}` , but it didnt work, i dont know why. Then I decided to use `async` and seems it worked!